google_guest_agent/addresses.go

```diff
@@ -552,31 +540,24 @@ func enableNetworkInterfaces(ctx context.Context, config *cfg.Sections) error {
 		}
 		fallthrough
 	default:
-		dhcpCommand := config.NetworkInterfaces.DHCPCommand
-		if dhcpCommand != "" {
-			tokens := strings.Split(dhcpCommand, " ")
-			return run.Quiet(ctx, tokens[0], tokens[1:]...)
-		}
-
-		// Try IPv4 first as it's higher priority.
-		if err := run.Quiet(ctx, "dhclient", googleInterfaces...); err != nil {
-			return err
+		primaryInterface, err := network.GetInterfaceByMAC(newMetadata.Instance.NetworkInterfaces[0].Mac)
+		if err != nil {
+			return fmt.Errorf("error getting primary interface: %s", err)
 		}
+		manager, err := network.DetectNetworkManager(ctx, primaryInterface.Name)
 
-		if len(googleIpv6Interfaces) == 0 {
-			return nil
-		}
-		for _, iface := range googleIpv6Interfaces {
-			// Enable kernel to accept to route advertisements.
-			val := fmt.Sprintf("net.ipv6.conf.%s.accept_ra_rt_info_max_plen=128", iface)
-			if err := run.Quiet(ctx, "sysctl", val); err != nil {
-				return err
+		if manager != currManagerService {
+			if err = currManagerService.Rollback(ctx, newMetadata.Instance.NetworkInterfaces); err != nil {
```

Comment:
you only want to Rollback if the `currManager` is no `nil`.

Reply:
Done.